pkg/parser/types/field_type.go

```diff
@@ -40,6 +40,21 @@ var (
 	TiDBStrictIntegerDisplayWidth bool
 )
 
+// IHasher is internal usage represent cascades/base.Hasher
+type IHasher interface {
```

Comment:
parser pkg sub-mod couldn't import back tidb pkg, which is undesirable, while for base.hasher, we can define the same even anonimous interface with same func signatures

Reply:
then, when calling from tidb mod, we can directly pass the base.Hasher as it's param here, implicit convertion will be conducted inside.